### DISCUSSION TITLE
How do branches work?

### DISCUSSION THREAD

**Discussion by starthlo** [gptme/gptme#136] [Q&A] [answered]
Hi, Could you explain about branches concept?

**Comment by ErikBjare** [comment, marked as answer]
The idea was to not delete history when users choose to edit the conversation history with `/edit`, but instead move the old conversation into a "branch" and make the edited conversation the new main branch.

If you have a look at the conversation directory after you've forked a conversation with `/fork` or `/edit`, it might make more sense to you:

```sh
$ tree ~/.local/share/gptme/logs/2024-09-24-laughing-happy-bird/
├── branches
│   └── main-edit-0.jsonl
└── conversation.jsonl
```

**Comment by starthlo** [comment]
I understand. Thank you for your time.

**Reply by ErikBjare** [reply]
Happy to clarify :)
I just updated it with an example, in case you missed it.
